### PR TITLE
fix: stack.slice not a function

### PR DESCRIPTION
**Pull request by kirrg001**
Reported by customer.

### CODE REVIEW

File: packages/core/src/util/stackTrace.js [hunk 1]
```diff
@@ -37,11 +37,17 @@ exports.captureStackTrace = function captureStackTrace(length, referenceFunction
...
+
+  if (!Array.isArray(stack)) {
+    stack = [];
+  }
+
   Error.stackTraceLimit = originalLimit;
   Error.prepareStackTrace = originalPrepareStackTrace;
 
-  if (drop > 0) {
+  if (drop > 0 && stack.length >= drop) {
```
Comment (kirrg001) [inline packages/core/src/util/stackTrace.js:50]: IMO there were two bugs.

1. stack not being a string (the actual use case is not 100% known yet)
2. stack.length >= drop behaviour

Comment (basti1302) [inline packages/core/src/util/stackTrace.js:50]: _praise:_ good catch, that way we at least get some stack trace if `drop > stack.length` accidentally. (I don't think this should ever happen, but anyway.)

**Review by basti1302 — APPROVED** [review]

File: packages/core/src/util/stackTrace.js (hunk 1 above)
Comment (basti1302) [reply packages/core/src/util/stackTrace.js:50]: I wonder when (2) would have happened in practice (without the new stack = [] fix)

**Review by aryamohanan — COMMENTED** [review]
Code looks good to me 👍

File: packages/core/src/util/stackTrace.js (hunk 1 above)
Comment (kirrg001) [reply packages/core/src/util/stackTrace.js:50]: Yeah 👍

Comment (kirrg001) [reply packages/core/src/util/stackTrace.js:50]: I am not sure tbh, I just looked for potential bugs in this fn :)

**Pull request merged**